src/test/compile-fail/generator/borrowing.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
Can we have UI tests? This is a new feature with interesting error UI, and I'll like to see it.